src/js/lib/content/src/index.ts

```diff
@@ -566,21 +667,104 @@ function tryAsString(value: any): string {
     return value.toString()
 }
 
-/// Main entry point. Loads WASM, initializes it, chooses the scene to run.
-API.main = async function (inputConfig: any) {
-    const urlParams = new URLSearchParams(window.location.search)
-    // @ts-ignore
-    const urlConfig = Object.fromEntries(urlParams.entries())
+async function fetchApplicationConfig(config: Config) {
+    const https = require('https')
 
-    const config = Config.default()
-    config.updateFromObject(inputConfig)
-    config.updateFromObject(urlConfig)
+    return new Promise((resolve: any, reject: any) => {
+        https.get(config.application_config_url, (res: any) => {
+            if (res.statusCode !== 200) {
+                reject(
+                    new Error(`Request ${config.application_config_url} return ${res.statusCode}.`)
+                )
+            }
+
+            res.setEncoding('utf8')
+            let rawData = ''
+
+            res.on('data', (chunk: any) => {
+                rawData += chunk
+            })
 
+            res.on('end', () => {
+                try {
+                    const parsedData = JSON.parse(rawData)
+                    resolve(parsedData)
+                } catch (e) {
+                    reject(e)
+                }
+            })
+
+            res.on('error', (e: any) => {
+                reject(e)
+            })
```

Comment:
you're using too many curly braces here and there. Please clean that in the code. For example, this can be rewritten to `res.on('error', (e: any) => reject(e))`